api/v1/inferencepool_types.go

```diff
@@ -66,17 +66,26 @@ type InferencePoolSpec struct {
 	// +required
 	Selector LabelSelector `json:"selector,omitempty,omitzero"`
 
-	// TargetPortNumber defines the port number to access the selected model server Pods.
-	// The number must be in the range 1 to 65535.
-	//
-	// +kubebuilder:validation:Minimum=1
-	// +kubebuilder:validation:Maximum=65535
+	// TargetPorts defines a list of ports that are exposed by this InferencePool.
+	// Currently, the list may only include a single port definition.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	// +listType=map
+	// +listMapKey=number
 	// +required
-	TargetPortNumber int32 `json:"targetPortNumber,omitempty"`
+	TargetPorts []Port `json:"targetPorts,omitempty"`
 
```

Comment:
@robscott if this field is required, we cannot support the use case of running the EPP as a sidecar without making API changes. Otherwise, we make `ExtensionRef` optional and add godocs that state something like:

```go
// When unspecified, the implementation MUST have the necessary information
// required to connect to the extension, handle failover, etc.
```

cc: @nirrozenbaum @kfswain

Reply:
We can always loosen validation in a future release, but tightening validation would be a backwards incompatible change. Is it OK if we leave this for a future release when the sidecar implementation details are clearer?

Reply:
I change `ExtensionRef` to be optional